components/psi/src/main/kotlin/io/github/composegears/valkyrie/psi/IconPackPsiParser.kt

```diff
@@ -0,0 +1,65 @@
+package io.github.composegears.valkyrie.psi
+
+import io.github.composegears.valkyrie.extensions.castOrNull
+import java.nio.file.Path
+import kotlin.io.path.name
+import kotlin.io.path.readText
+import org.jetbrains.kotlin.com.intellij.psi.PsiManager
+import org.jetbrains.kotlin.com.intellij.testFramework.LightVirtualFile
+import org.jetbrains.kotlin.idea.KotlinFileType
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtObjectDeclaration
+import org.jetbrains.kotlin.psi.KtTreeVisitorVoid
+
+data class IconPackInfo(
+    val packageName: String,
+    val iconPack: String,
+    val nestedPacks: List<String>,
+)
+
+object IconPackPsiParser {
+
+    fun extractIconPack(path: Path): IconPackInfo? {
+        val ktFile = PsiManager.getInstance(project)
+            .findFile(
+                LightVirtualFile(
+                    /* name = */
+                    path.name,
+                    /* fileType = */
+                    KotlinFileType.INSTANCE,
+                    /* text = */
```

Comment:
These comments could be replaced with field names?

Reply:
`LightVirtualFile` is java class 🙁 
will remove comments, looks redundant